Tools/CISupport/built-product-archive

```diff
@@ -182,41 +195,20 @@ def dirContainsdwo(directory):
     return False
 
 
-def minifyDirectory(platform, directory):
-    if platform in ['mac', 'ios', 'tvos', 'watchos']:
-        dir_patterns = ('.dSYM', 'DerivedSources')
-        file_patterns = ('.a',)
-        for root, dirs, files in os.walk(directory):
-            if any(root.endswith(pattern) for pattern in dir_patterns):
-                print('Removing: {}'.format(root))
-                shutil.rmtree(root)
-            for name in files:
-                if any(name.endswith(pattern) for pattern in file_patterns):
-                    print('Removing: {}'.format(os.path.join(root, name)))
-                    os.remove(os.path.join(root, name))
-
-
 def archiveBuiltProduct(configuration, platform, fullPlatform, minify=False):
     assert platform in ('gtk', 'ios', 'jsc', 'mac', 'tvos', 'watchos', 'win', 'wincairo', 'wpe')
     global _configurationBuildDirectory
 
     if platform in ['ios', 'tvos', 'watchos']:
-        combinedDirectory = os.path.join(_topLevelBuildDirectory, 'combined-mac-and-{}'.format(platform))
-        removeDirectoryIfExists(combinedDirectory)
-        os.makedirs(combinedDirectory)
         if minify:
-            minifyDirectory(platform, _configurationBuildDirectory)
-            minifyDirectory(platform, _hostBuildDirectory)
-        if subprocess.call(['/bin/cp', '-pR', _configurationBuildDirectory, combinedDirectory]):
-            return 1
-        if subprocess.call(['/bin/cp', '-pR', _hostBuildDirectory, combinedDirectory]):
-            return 1
-
-        return createZip(combinedDirectory, configuration, minify=minify)
+            return createZip([_configurationBuildDirectory, _hostBuildDirectory], 'minified-' + configuration, excludePatterns=('*.a', '*.dSYM', 'DerivedSources'))
+        else:
+            return createZip([_configurationBuildDirectory, _hostBuildDirectory], configuration)
     elif platform == 'mac':
         if minify:
-            minifyDirectory(platform, _configurationBuildDirectory)
-        return createZip(_configurationBuildDirectory, configuration, embedParentDirectoryNameOnDarwin=True, minify=minify)
+            return createZip([_configurationBuildDirectory], 'minified-' + configuration, excludePatterns=('*.a', '*.dSYM', 'DerivedSources'), embedParentDirectoryNameOnDarwin=True)
```

Comment:
I'm usually not the one to ask for extracting constants and putting them at the top of a file, but something like this (especially repeated) seems like a good thing to put at the top level somewhere, so that we could reason about archive contents and about how the code is written separately.